processor/error/processor.go

```diff
@@ -20,12 +20,12 @@ const (
 func NewProcessor() pr.Processor {
 	schema := pr.CreateSchema(errorSchema, processorName)
 	return &processor{
-		&Payload{},
+		&payload{},
 		schema}
 }
 
 type processor struct {
-	payload *Payload
+	payload pr.Payload
```

Comment:
Why did you change this? I think it's good to depend on the local payload. We should change it to the same as we have in the `transaction` processor.